### PR TITLE
Added error messages to the Lark LaTeX parser

### PR DESCRIPTION
**Pull request by wermos**
#### Brief description of what is fixed or changed

When writing the LaTeX parser, I had left most of the error messages as TODOs. In this PR, I filled out all the error messages with helpful strings.

#### Other comments

cc @sylee957 and @Upabjojr for review.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/parsing/latex/lark/transformer.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ def relation(self, tokens):
         elif relation_type == "GTE":
             return sympy.Ge(tokens[0], tokens[2])
         else:
-            raise LaTeXParsingError() # TODO: Fill descriptive error message.
+            raise LaTeXParsingError("An unsupported relational symbol was found.")
```
Comment (sylee957) [inline sympy/parsing/latex/lark/transformer.py:101]: I'm not sure if the rule is really reachable, this already exhausts all tokens, 
so the error should be meaningful for some future

`relation: _expression (EQUAL | NOT_EQUAL | LT | LTE | GT | GTE) _expression`

Comment (wermos) [reply sympy/parsing/latex/lark/transformer.py:101]: That was what I was going for. Can you give me some suggestions on how to improve the error message?

Comment (sylee957) [reply sympy/parsing/latex/lark/transformer.py:101]: I may suggest in other ways to organize the grammar
from
```lark
relation: _expression (EQUAL | NOT_EQUAL | LT | LTE | GT | GTE) _expression
```
to
```lark
relation: eq | ne | lt | le | gt | ge
eq: _expression EQUAL _expression
ne: _expression EQUAL _expression
...
```

which eliminates the need to write the if-statements inside transformers.

Comment (wermos) [reply sympy/parsing/latex/lark/transformer.py:101]: That sounds good to me. I'm currently finishing up a PR which adds derivative support, among many other things. I'll make this change in that PR.

**Pull request merged**